nautobot/ipam/models.py

```diff
@@ -1079,30 +1089,31 @@ def clean(self):
         if self.type == choices.IPAddressTypeChoices.TYPE_SLAAC and self.ip_version != 6:
             raise ValidationError({"type": "Only IPv6 addresses can be assigned SLAAC type"})
 
-        # Force dns_name to lowercase
-        self.dns_name = self.dns_name.lower()
+        # If neither `parent` or `namespace` was provided; raise this exception
+        if self._namespace is None:
+            raise ValidationError({"parent": "Namespace could not be determined."})
 
-    def save(self, *args, **kwargs):
-        if not self.present_in_database:
-            if self._namespace is None:
-                raise ValidationError({"parent": "Namespace could not be determined."})
-            namespace = self._namespace
-        else:
-            namespace = self._namespace or self.parent.namespace
-
-        # Determine the closest parent automatically based on the Namespace.
-        self.parent = (
-            Prefix.objects.filter(namespace=namespace)
-            # 3.0 TODO: disallow IPAddress from parenting to a TYPE_POOL prefix, instead pick closest TYPE_NETWORK
-            # .exclude(type=choices.PrefixTypeChoices.TYPE_POOL)
-            .get_closest_parent(self.host, include_self=True)
-        )
+        # Validate `parent` can be used as the parent for this ipaddress
+        if self.parent:
+            closest_parent = self._get_closest_parent()
+            if self.parent != closest_parent:
+                raise ValidationError({"parent": f"{self.parent} cannot be assigned as a parent."})
```

Comment:
Perhaps something more explicit? Like:

```suggestion
                raise ValidationError(
                    {
                        "parent": (
                            f"{self.parent} cannot be assigned as the parent of {self}. "
                            f" In namespace {self._namespace}, the expected parent would be {closest_parent}."
                        )
                    }
                )
```